File: VB/BlazorServerApp/Controllers/CultureController.vb
Imports Microsoft.AspNetCore.Localization
Imports Microsoft.AspNetCore.Mvc

Namespace BlazorServerApp.Controllers

    <Route("[controller]/[action]")>
    Public Class CultureController
        Inherits Controller

        Public Function SetCulture(ByVal culture As String, ByVal redirectUri As String) As IActionResult
            If Not Equals(culture, Nothing) Then
                HttpContext.Response.Cookies.Append(CookieRequestCultureProvider.DefaultCookieName, CookieRequestCultureProvider.MakeCookieValue(New RequestCulture(culture)))
            End If

            Return LocalRedirect(redirectUri)
        End Function
    End Class
End Namespace

File: CS/DxBlazorLocalization/BlazorServer/Controllers/CultureController.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Localization;

namespace BlazorServer.Controllers {
    [Route("[controller]/[action]")]
    public class CultureController : Controller {
        public IActionResult Set(string culture, string redirectUri) {
            if (culture != null) {
                HttpContext.Response.Cookies.Append(
                    CookieRequestCultureProvider.DefaultCookieName,
                    CookieRequestCultureProvider.MakeCookieValue(
                        new RequestCulture(culture, culture)));
            }
            return LocalRedirect(redirectUri);
        }
    }
}
